From b0a8c85741c828eb8d82da17a49876176e120a08 Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Fri, 18 Jun 2021 16:42:27 +0300
Subject: [PATCH 15/29] include: s32gen1: Add recovery pinmuxing for i2c

Add recovery pinmuxing definition for ERR050686 "I2C: I2C controller
is unable to generate clocks when SDA is low".

Upstream-Status: Pending

Issue: ALB-7193
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/dt-bindings/pinctrl/s32-gen1-pinctrl.h |  5 +++++
 include/dt-bindings/pinctrl/s32g-pinctrl.h     | 12 ++++++++++++
 include/dt-bindings/pinctrl/s32r45-pinctrl.h   |  4 ++++
 3 files changed, 21 insertions(+)

diff --git a/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h b/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
index dd3cc097d3..174918509e 100644
--- a/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32-gen1-pinctrl.h
@@ -32,6 +32,8 @@
 #define SIUL2_MSCR_S32_G1_MUX_ID_6	(6)
 #define SIUL2_MSCR_S32_G1_MUX_ID_7	(7)
 
+#define IMCR_DISABLED		SIUL2_MSCR_S32_G1_MUX_ID_0
+
 #define DSPI_SCK_PIN_CFG	(SIUL2_MSCR_S32_G1_OBE | \
 	SIUL2_MSCR_S32_G1_SRC_50MHz)
 
@@ -89,6 +91,9 @@
 #define USB_ULPI_DATA_PIN_CFG	(SIUL2_MSCR_S32_G1_SRC_208MHz | \
 	 SIUL2_MSCR_S32_G1_OBE | SIUL2_MSCR_S32_G1_IBE)
 
+#define	SIUL2_MSCR_GPIO		(SIUL2_MSCR_S32_G1_MUX_ID_0 | \
+	SIUL2_MSCR_S32_G1_ODE | SIUL2_MSCR_S32_G1_OBE)
+
 #define SD0_CMD_IMCR		(515)
 #define SD0_D0_IMCR		(516)
 #define SD0_D1_IMCR		(517)
diff --git a/include/dt-bindings/pinctrl/s32g-pinctrl.h b/include/dt-bindings/pinctrl/s32g-pinctrl.h
index 10906a185f..a70903966f 100644
--- a/include/dt-bindings/pinctrl/s32g-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32g-pinctrl.h
@@ -113,36 +113,42 @@
 #define PB00_I2C0_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB00_I2C0_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB00_I2C0_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB01 */
 #define PB01_MSCR_S32G		(17)
 #define PB01_I2C0_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB01_I2C0_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB01_I2C0_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB03 */
 #define PB03_MSCR_S32G		(19)
 #define PB03_I2C1_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB03_I2C1_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB03_I2C1_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB04 */
 #define PB04_MSCR_S32G		(20)
 #define PB04_I2C1_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB04_I2C1_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB04_I2C1_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB05 */
 #define PB05_MSCR_S32G		(21)
 #define PB05_I2C2_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB05_I2C2_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB05_I2C2_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB06 */
 #define PB06_MSCR_S32G		(22)
 #define PB06_I2C2_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB06_I2C2_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB06_I2C2_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB09 */
 #define PB09_MSCR_S32G		(25)
@@ -159,24 +165,28 @@
 #define PB15_I2C0_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_2 | \
 	I2C_PIN_CFG)
 #define PB15_I2C0_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_3)
+#define PB15_I2C0_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PC00 */
 #define PC00_MSCR_S32G		(32)
 #define PC00_I2C0_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PC00_I2C0_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_3)
+#define PC00_I2C0_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PC01 */
 #define PC01_MSCR_S32G		(33)
 #define PC01_I2C4_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PC01_I2C4_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_3)
+#define PC01_I2C4_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PC02 */
 #define PC02_MSCR_S32G		(34)
 #define PC02_I2C4_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_2 | \
 	I2C_PIN_CFG)
 #define PC02_I2C4_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_3)
+#define PC02_I2C4_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PC14 */
 #define PC14_MSCR_S32G		(46)
@@ -586,12 +596,14 @@
 #define PK03_I2C1_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_3 | \
 	I2C_PIN_CFG)
 #define PK03_I2C1_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_5)
+#define PK03_I2C1_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PK05 */
 #define PK05_MSCR_S32G		(165)
 #define PK05_I2C1_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_3 | \
 	I2C_PIN_CFG)
 #define PK05_I2C1_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_4)
+#define PK05_I2C1_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PL08 */
 #define PL08_MSCR_S32G		(184)
diff --git a/include/dt-bindings/pinctrl/s32r45-pinctrl.h b/include/dt-bindings/pinctrl/s32r45-pinctrl.h
index 32bc8f260c..af9da3c2eb 100644
--- a/include/dt-bindings/pinctrl/s32r45-pinctrl.h
+++ b/include/dt-bindings/pinctrl/s32r45-pinctrl.h
@@ -21,24 +21,28 @@
 #define PA14_I2C1_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_4 | \
 	I2C_PIN_CFG)
 #define PA14_I2C1_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PA14_I2C1_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PA15 */
 #define PA15_MSCR_S32R45	(15)
 #define PA15_I2C1_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_3 | \
 	I2C_PIN_CFG)
 #define PA15_I2C1_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PA15_I2C1_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB00 */
 #define PB00_MSCR_S32R45	(16)
 #define PB00_I2C0_SDA_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB00_I2C0_SDA_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB00_I2C0_SDA_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB01 */
 #define PB01_MSCR_S32R45	(17)
 #define PB01_I2C0_SCL_CFG	(SIUL2_MSCR_S32_G1_MUX_ID_1 | \
 	I2C_PIN_CFG)
 #define PB01_I2C0_SCL_IN	(SIUL2_MSCR_S32_G1_MUX_ID_2)
+#define PB01_I2C0_SCL_GPIO	(SIUL2_MSCR_GPIO)
 
 /* PB03 */
 #define PB03_MSCR_S32R45	(19)
-- 
2.17.1

